starlette/formparsers.py

```diff
@@ -247,21 +249,28 @@ async def parse(self) -> FormData:
 
         # Create the parser.
         parser = multipart.MultipartParser(boundary, callbacks)
-        # Feed the parser with data from the request.
-        async for chunk in self.stream:
-            parser.write(chunk)
-            # Write file data, it needs to use await with the UploadFile methods that
-            # call the corresponding file methods *in a threadpool*, otherwise, if
-            # they were called directly in the callback methods above (regular,
-            # non-async functions), that would block the event loop in the main thread.
-            for part, data in self._file_parts_to_write:
-                assert part.file  # for type checkers
-                await part.file.write(data)
-            for part in self._file_parts_to_finish:
-                assert part.file  # for type checkers
-                await part.file.seek(0)
-            self._file_parts_to_write.clear()
-            self._file_parts_to_finish.clear()
+        try:
+            # Feed the parser with data from the request.
+            async for chunk in self.stream:
+                parser.write(chunk)
+                # Write file data, it needs to use await with the UploadFile methods
+                # that call the corresponding file methods *in a threadpool*,
+                # otherwise, if they were called directly in the callback methods above
+                # (regular, non-async functions), that would block the event loop in
+                # the main thread.
+                for part, data in self._file_parts_to_write:
+                    assert part.file  # for type checkers
+                    await part.file.write(data)
+                for part in self._file_parts_to_finish:
+                    assert part.file  # for type checkers
+                    await part.file.seek(0)
+                self._file_parts_to_write.clear()
+                self._file_parts_to_finish.clear()
+        except Exception as e:
+            # Close all the files if there was an error.
+            for file in self._files_to_close_on_error:
+                file.close()
+            raise e
```

Comment:
Just for consistency
```suggestion
            raise exc
```